homeassistant/components/vacuum/__init__.py

```diff
@@ -173,36 +175,51 @@ class _BaseVacuum(Entity):
     Contains common properties and functions for all vacuum devices.
     """
 
+    _attr_battery_icon: str | None = None
```

Comment:
I am not sure if it is useful to implement `_attr_battery_icon` since it is overriden in both `VacuumEntity` and `StateVacuumEntity`

Reply:
It can be useful, or, if specific needs, implement it in the child entities